src/modeler/AutoRest.Swagger/Validation/Core/JsonValidationLogListener.cs

```diff
@@ -29,7 +29,23 @@ public void Log(LogMessage message)
                     .FirstOrDefault();
                 var pathComponents = resPathPattern.Match(path ?? "");
                 var pathComponentProviderNamespace = pathComponents.Groups["providerNamespace"];
-                var pathComponentResourceType = pathComponents.Groups["resourceType"];
+                var pathComponentResource = pathComponents.Groups["resource"];
```

Comment:
Or, use the same regex and check the number of groups returned by regex and accordingly assign, parent, child or grandchild